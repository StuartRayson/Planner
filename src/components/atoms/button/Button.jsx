import classNames from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import './button.scss';

const Button = (props) => {
  const {
    variant = 'primary',
    href,
    disabled,
    onClick,
    children,
    className } = props;
  const TagName = href ? 'a' : 'button';
  const classes = classNames(
    'a-button',
    `a-button--${variant}`,
    disabled && 'a-button--disabled',
    className
  );

  const attributeOverrides = {
    variant: undefined,
    disabled: (TagName === 'button') && disabled,
    onClick: (TagName === 'button' && onClick) ? onClick : undefined
  };

  return (
    <TagName {...props} {...attributeOverrides} className={classes}>
      {children}
    </TagName>
  );
};

Button.propTypes = {
  className: PropTypes.string,
  type: PropTypes.string,
  variant: PropTypes.string,
  disabled: PropTypes.bool,
  children: PropTypes.node,
  href: PropTypes.string,
  onClick: PropTypes.func
};

export default Button;
