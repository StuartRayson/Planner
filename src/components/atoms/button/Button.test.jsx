import React from 'react';
import { shallow } from 'enzyme';

import Button from './Button.jsx';

describe('<Button />', () => {
  it('renders without crashing', () => {
    shallow(<Button />);
  });

  it('simulates click events', () => {
    const onButtonClick = jest.fn();
    const button = shallow(<Button onClick={onButtonClick} />);
    button.find('button').simulate('click');
    expect(onButtonClick.mock.calls.length).toEqual(1);
  });

  it('should render anchor tag', () => {
    const button = shallow(<Button href="test" />);
    expect(button.find('a').length).toEqual(1);
  });

  it('should render inner text', () => {
    const button = shallow(<Button>Submit</Button>);
    expect(button.text()).toEqual('Submit');
  });

  it('should be disabled', () => {
    const button = shallow(<Button disabled>Submit</Button>);
    expect(button.find('button').prop('disabled')).toEqual(true);
  });

  it('should renders button correctly', () => {
    const wrapper = shallow(<Button>Submit</Button>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should renders anchor tag correctly', () => {
    const wrapper = shallow(<Button href="/link">Link</Button>);
    expect(wrapper).toMatchSnapshot();
  });
});
