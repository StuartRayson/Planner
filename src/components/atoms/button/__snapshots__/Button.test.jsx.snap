// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should renders anchor tag correctly 1`] = `
<a
  className="a-button a-button--primary"
  disabled={false}
  href="/link"
>
  Link
</a>
`;

exports[`<Button /> should renders button correctly 1`] = `
<button
  className="a-button a-button--primary"
>
  Submit
</button>
`;
